### ISSUE TITLE
Showing the DataGrid column header separators requires toggling AreSeparatorsVisible off and on to enable

### ISSUE THREAD

**Issue by garyhertel** [AvaloniaUI/Avalonia#2575]
Getting the DataGrid column header separators to show requires setting the `AreSeparatorsVisible` to `false` and then `true`. 

It looks like this is due to the `AreSeparatorsVisible` being initialized to `true` and the underlying field `_desiredSeparatorVisibility` being initialized to `false`. There's also an is changed check when setting the `AreSeparatorsVisible` so it needs to get set to `false` and then `true` to trigger.

Here's what I eventually did to get this working:
```
  <Style Selector="dg|DataGridColumnHeader">
      <Setter Property="AreSeparatorsVisible" Value="false" />
      <Setter Property="AreSeparatorsVisible" Value="true" />
  </Style>

  <Style Selector="dg|DataGridColumnHeader /template/ Rectangle#VerticalSeparator">
    <Setter Property="Stroke" Value="Black"/>
    <Setter Property="StrokeThickness" Value="1"/>
  </Style>
```
It seems like these separators should be enabled by default so users don't have to guess where the separator is when resizing too.

**Comment by garyhertel** [comment]
One other issue I noticed is there's no separator on the right side of the last column if it doesn't fill up the available width. This gives it a very odd and inconsistent look. (most datagrids including WPF will show this separator).

**Comment by maxkatz6** [comment]
Fixed in 0.10 preview
https://github.com/AvaloniaUI/Avalonia/pull/4534/files#diff-c4c53d1bd698d898bf493b888d480f25R43
